src/NCalc.Async/AsyncExpression.cs

```diff
@@ -242,55 +242,41 @@ public bool HasErrors()
     {
         var parameterEnumerators = new Dictionary<string, IEnumerator>();
         int? size = null;
-        var results = new List<object?>();
 
-        foreach (var parameter in Parameters.Values)
+        foreach (var parameter in Parameters)
         {
-            if (parameter is not IEnumerable enumerable)
-                continue;
+            if (parameter.Value is IEnumerable enumerable)
+            {
+                parameterEnumerators.Add(parameter.Key, enumerable.GetEnumerator());
 
-            var localsize = enumerable.Cast<object>().Count();
+                int localSize = 0;
+                foreach (var o in enumerable)
```

Comment:
Possible multiple enumeration warning here
https://www.jetbrains.com/help/rider/PossibleMultipleEnumeration.html

Reply:
Thanks, I'll check it later, need to benchmark casting vs iterating

Reply:
I did a benchmark with 1000 elements and casting is really faster, I'll rollback my change:

```
| Method      | Mean        | Error      | StdDev     | Rank | Allocated |
|------------ |------------:|-----------:|-----------:|-----:|----------:|
| TestCast    |    15.20 ns |   0.220 ns |   0.253 ns |    1 |         - |
| TestIterate | 8,159.16 ns | 160.834 ns | 225.467 ns |    2 |      40 B |
```

Reply:
I think just a `ToList` is missing at `var localsize = enumerable.Cast<object>().Count();` to solve the warning

Reply:
I think it's a false positive, just reverted casting

Reply:
No, I agree with the warning, it's possible to solve 2 warnings this way:

![image](https://github.com/ncalc/ncalc/assets/52143624/12cb64a4-18a0-408b-8982-a9c112c9d4da)
![image](https://github.com/ncalc/ncalc/assets/52143624/cbeed473-7942-489f-b63a-c6b8597cad51)


![image](https://github.com/ncalc/ncalc/assets/52143624/5860b111-14a9-46ba-ac6b-3fb58c04af20)

Reply:
Agreed, as we're looping through iterator it'll automatically disposed, but still there is a case when the size of arrays are not equal and we still need to dispose enumerator, isn't it?

Reply:
No, the dispose is because the type of the `IEnumerator` is not known. With a `List` we know it never needs to be disposed.